### PR TITLE
Add a CRUDStoreAdapter specifically for encryptable models

### PR DESCRIPTION
**Pull request by GabeLambda**
Currently only adds an override to support encrypted model re-encryption
(e.g. for re-encrypting data using a separate master key).

I haven't yet added any tests, I wanted feedback on basic structure before I started checking anything here.

### CODE REVIEW

File: microcosm_flask/encryption/conventions/crud_adapter.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        """
+        encrypted_field_name = self.store.model_class.__plaintext__
+
+        id_ = kwargs[self.identifier_key]
+        current_model = self.store.retrieve(id_)
+        current_value = current_model.plaintext
+
+        null_update = (
+            # Check if the update is for the encrypted field, and if it's explicitly set to null
```
Comment (pelegdvir) [inline microcosm_flask/encryption/conventions/crud_adapter.py]: That's going to be problematic  - with default values:
```
update(self, foo=None, **kwargs):
    return super().update(foo=foo, **kwargs)
```

File: microcosm_flask/encryption/conventions/crud_adapter.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from microcosm_flask.conventions.crud_adapter import CRUDStoreAdapter
+
+
+class EncryptableCRUDStoreAdapter(CRUDStoreAdapter):
+    """
+    Adapt the CRUD conventions callbacks to the `EncryptableStore` interface.
+
+    """
+
+    def update(self, **kwargs):
```
Comment (pelegdvir) [inline microcosm_flask/encryption/conventions/crud_adapter.py]: Maybe it's preferable to call this: update_and_reencrypt.
In some cases, we want to use the update function again and again without reencrypting the data.

Comment (GabeLambda) [reply microcosm_flask/encryption/conventions/crud_adapter.py]: Okay.

File: microcosm_flask/encryption/conventions/crud_adapter.py (hunk 1 above)
Comment (GabeLambda) [reply microcosm_flask/encryption/conventions/crud_adapter.py]: I'd argue that updates shouldn't ever pass in default values. You should only be specifying exactly what you want to update here.

File: microcosm_flask/tests/encryption/conventions/fixtures/encryptable_models.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from uuid import UUID
+
+
+@dataclass
+class Encryptable:
+    id: UUID
+    value: Optional[str] = None
+    other_value: Optional[str] = None
+
+    # # encryption support
```
Comment (pelegdvir) [inline microcosm_flask/tests/encryption/conventions/fixtures/encryptable_models.py]: ?

**Pull request merged**